core/unit_test/view/TestReferenceCountedAccessor.hpp

```diff
@@ -0,0 +1,156 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+
+#include <gtest/gtest.h>
+
+namespace {
+using element_t      = float;
+using mem_t          = typename TEST_EXECSPACE::memory_space;
+using defacc_t       = Kokkos::default_accessor<element_t>;
+using const_defacc_t = Kokkos::default_accessor<const element_t>;
+using acc_t =
+    Kokkos::Impl::ReferenceCountedAccessor<element_t, mem_t, defacc_t>;
+using const_acc_t =
+    Kokkos::Impl::ReferenceCountedAccessor<const element_t, mem_t,
+                                           const_defacc_t>;
+using data_handle_t       = typename acc_t::data_handle_type;
+using const_data_handle_t = typename const_acc_t::data_handle_type;
+}  // namespace
+
+void test_refcountedacc_typedefs() {
+  static_assert(std::is_same_v<typename acc_t::element_type, element_t>);
+  static_assert(std::is_same_v<
+                typename acc_t::data_handle_type,
+                Kokkos::Impl::ReferenceCountedDataHandle<element_t, mem_t>>);
+  static_assert(
+      std::is_same_v<typename acc_t::reference, typename defacc_t::reference>);
+  static_assert(
+      std::is_same_v<typename acc_t::offset_policy,
+                     Kokkos::Impl::ReferenceCountedAccessor<
+                         element_t, mem_t, typename defacc_t::offset_policy>>);
+}
+
+TEST(TEST_CATEGORY, RefCountedAcc_Typedefs) { test_refcountedacc_typedefs(); }
+
+template <class T>
+KOKKOS_FUNCTION void unused_variable_sink(T) {}
+
+void test_refcountedacc_ctors() {
+  Kokkos::parallel_for(Kokkos::RangePolicy<TEST_EXECSPACE>(0, 1), KOKKOS_LAMBDA(int) {
+      // default ctor and non-const to const
+      {
+        acc_t acc;
+        const_acc_t c_acc(acc);
+	static_assert(!std::is_constructible_v<acc_t, const_acc_t>);
+
+	unused_variable_sink(c_acc);
+}
+// from default_accessor
+{
+  defacc_t defacc;
+  const_defacc_t c_defacc;
+  acc_t acc(defacc);
+  const_acc_t c_acc1(defacc);
+  const_acc_t c_acc2(c_defacc);
+  static_assert(!std::is_constructible_v<acc_t, const_defacc_t>);
+
+  unused_variable_sink(acc);
+  unused_variable_sink(c_acc1);
+  unused_variable_sink(c_acc2);
+}
+});
```

Comment:
```suggestion
  Kokkos::parallel_for(Kokkos::RangePolicy<TEST_EXECSPACE>(0, 1), KOKKOS_LAMBDA(int) {
    // default ctor and non-const to const
    {
      acc_t acc;
      const_acc_t c_acc(acc);
      static_assert(!std::is_constructible_v<acc_t, const_acc_t>);

      unused_variable_sink(c_acc);
    }
    // from default_accessor
    {
      defacc_t defacc;
      const_defacc_t c_defacc;
      acc_t acc(defacc);
      const_acc_t c_acc1(defacc);
      const_acc_t c_acc2(c_defacc);
      static_assert(!std::is_constructible_v<acc_t, const_defacc_t>);

      unused_variable_sink(acc);
      unused_variable_sink(c_acc1);
      unused_variable_sink(c_acc2);
    }
  });
```

I think i got it right, no idea why clang-format wasn't complaining about this

Reply:
It's actually worse than that. It undoes these changes again. At least it did on kokkos-dev-2